Mobile Buy SDK/Mobile Buy SDK/Models/BUYManagedObject.m

```diff
@@ -91,6 +92,83 @@ - (NSDictionary *)jsonDictionaryForCheckout
 	return self.JSONDictionary;
 }
 
+#pragma mark - Dynamic Property Accessor Resolution
+
+NS_INLINE NSString *KeyForSelector(SEL selector, BOOL *isSetter)
+{
+	NSString *name = NSStringFromSelector(selector);
+	BOOL setter = [name hasPrefix:@"set"];
+	if (isSetter) {
+		*isSetter = setter;
+	}
+	return setter ? [[name substringWithRange:NSMakeRange(3, name.length - 4)] lowercaseString] : name;
+}
+
+- (BOOL)respondsToSelector:(SEL)selector
+{
+	return [super respondsToSelector:selector] || [self isPropertyAccessor:selector];
+}
+
+static NSMutableDictionary *signatureCache;
```

Comment:
I'm not really a fan of file variables. Would personally prefer to see this in a class method instead (but just my opinion):
```objc
+ (NSMutableDictionary *)signatureCache
{
    static NSMutableDictionary *cache = nil;
    static dispatch_once_t onceToken;
	dispatch_once(&onceToken, ^{
        cache = [NSMutableDictionary new];
	});
    return cache;
}
```